### ISSUE TITLE
Docker Hub image outdated

### ISSUE THREAD

**Issue by KetchupBomb** [sudoforge/docker-images#94]
Just pulled `sudoforge/murmur:latest` and it has an outdated `docker-entrypoint.sh`:

```
/etc/murmur # curl -skL https://raw.githubusercontent.com/sudoforge/docker-images/master/murmur/docker-entrypoint.sh > docker-entrypoint.sh
/etc/murmur # diff /usr/local/bin/docker-entrypoint.sh docker-entrypoint.sh
50,53c50,53
<     SSL_CERTFILE=/data/cert.pem
<     SSL_KEYFILE=/data/key.pem
<     SSL_CAFILE=/data/intermediate.pem
<     SSL_DHFILE=/data/dh.pem
---
>     SSL_CERTFILE=${MUMBLE_CERTFILE:-/data/cert.pem}
>     SSL_KEYFILE=${MUMBLE_KEYFILE:-/data/key.pem}
>     SSL_CAFILE=${MUMBLE_CAFILE:-/data/intermediate.pem}
>     SSL_DHFILE=${MUMBLE_DHFILE:-/data/dh.pem}
76d75
<     setVal welcometext '"'
78c77
<     echo -n "${parsedContent}\"" >> "${CONFIGFILE}"
---
>     setVal welcometext "\"$parsedContent\""
94c93,101
<     /opt/murmur/murmur.x86 -ini "${CONFIGFILE}" -supw $SUPERUSER_PASSWORD
---
>
>     # Using -supw currently throws a fatal error and exits the program. It has
>     # been fixed in the upstream project by commit d8203ba94 [1], after the
>     # current version (1.3.0). To bypass the issue and allow the entrypoint
>     # script to proceed, we are ignoring the erroneous exit code by executing
>     # `true` if the first invocation fails.
>     #
>     # [1]: https://github.com/mumble-voip/mumble/commit/d8203ba94d528b092e0ff5a52a51af28f8f592f1
>     /opt/murmur/murmur.x86 -ini "${CONFIGFILE}" -supw $SUPERUSER_PASSWORD || true
```

---

Do you have a mechanism that auto-updates images upon commit?

**Comment by sudoforge** [comment]
Hey! Thanks for spotting this and opening an issue. That is _really, really strange_. I re-built the image locally and notice a difference (although not on the entrypoint):

```
➜ container-diff diff remote://sudoforge/murmurd:1.3.0-3 daemon://sudoforge/murmurd:latest --type=history --type=file

-----File-----

These entries have been added to sudoforge/murmurd:1.3.0-3: None

These entries have been deleted from sudoforge/murmurd:1.3.0-3: None

These entries have been changed between sudoforge/murmurd:1.3.0-3 and sudoforge/murmurd:latest:
FILE               SIZE1        SIZE2
/etc/shadow        469B         469B


-----History-----

Docker history lines found only in sudoforge/murmurd:1.3.0-3: None

Docker history lines found only in sudoforge/murmurd:latest: None
```

Running a container and copying it to my local filesystem didn't show a difference between the entrypoint:

```
gh::sudoforge/docker-images [ master ]
➜ docker cp 6d861f3621a6:/usr/local/bin/docker-entrypoint.sh murmur/docker-entrypoint.sh


gh::sudoforge/docker-images [ master ]
➜ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```  

I've manually built a new release, `:1.3.0-4`, and updated the `:latest` tag. Can you confirm that the issue is no longer present for you?

---

> Do you have a mechanism that auto-updates images upon commit?

Internally, yes. I was waiting for GitHub Actions to set this up, but due to the low churn here, never got around to doing that even after my account gained access. Thanks for the push -- I'll move this to Github Actions today.

**Comment by sudoforge** [comment]
The difference between the two `/etc/shadow` images is just the UID for the `murmur` user:

```
➜ diff -ur shadow-3 shadow-4
--- shadow-3    2020-03-20 00:11:25.000000000 -0700
+++ shadow-4    2020-04-04 06:57:05.000000000 -0700
@@ -26,4 +26,4 @@
 smmsp:!::0:::::
 guest:!::0:::::
 nobody:!::0:::::
-murmur:!:18341:0:99999:7:::
+murmur:!:18356:0:99999:7:::
```

**Comment by sudoforge** [comment]
Hey! Thanks for spotting this and opening an issue. That is _really, really strange_. I re-built the image locally and notice a difference (although not on the entrypoint):

```
➜ container-diff diff remote://sudoforge/murmurd:1.3.0-3 daemon://sudoforge/murmurd:latest --type=history --type=file

-----File-----

These entries have been added to sudoforge/murmurd:1.3.0-3: None

These entries have been deleted from sudoforge/murmurd:1.3.0-3: None

These entries have been changed between sudoforge/murmurd:1.3.0-3 and sudoforge/murmurd:latest:
FILE               SIZE1        SIZE2
/etc/shadow        469B         469B


-----History-----

Docker history lines found only in sudoforge/murmurd:1.3.0-3: None

Docker history lines found only in sudoforge/murmurd:latest: None
```

This is simply the UID of the `murmur` user, and is expected:

```
➜ diff -ur shadow-from-remote-latest shadow-from-local-latest
--- shadow-3    2020-03-20 00:11:25.000000000 -0700
+++ shadow-4    2020-04-04 06:57:05.000000000 -0700
@@ -26,4 +26,4 @@
 smmsp:!::0:::::
 guest:!::0:::::
 nobody:!::0:::::
-murmur:!:18341:0:99999:7:::
+murmur:!:18356:0:99999:7:::
```

Running a container and copying it to my local filesystem didn't show a difference between the entrypoint:

```
gh::sudoforge/docker-images [ master ]
➜ docker cp 6d861f3621a6:/usr/local/bin/docker-entrypoint.sh murmur/docker-entrypoint.sh

gh::sudoforge/docker-images [ master ]
➜ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```  

I imagine that you might have an outdated `:latest`. Can you run `docker inspect sudoforge/murmurd:latest --format='{{ .Id }}'` and provide the output here?

---

> Do you have a mechanism that auto-updates images upon commit?

Internally, yes. I was waiting for GitHub Actions to set this up, but due to the low churn here, never got around to doing that even after my account gained access. Thanks for the push -- I'll move this to Github Actions today.

**Comment by sudoforge** [comment]
@KetchupBomb Is this still an issue for you?

**Comment by KetchupBomb** [comment]
Haha, I see the problem. You have two images, `sudoforge/murmur` and `sudoforge/murmurd`.

I've been using the former (which was last updated 5 months ago), while the later (which was updated 2 days ago) has the changes I've been after.

Your readme has `sudoforge/murmur` in header text, but the body of the document uses the proper `sudoforge/murmurd`, I just didn't notice.

I leave it up to you to do what you wish. Marking as closed.

**Comment by sudoforge** [comment]
**Ah, you're totally right**

I can't recall why the image name was changed -- I don't have an issue describing the change; [just one referencing the new name from last year][0]... Shoot. I'm sorry for the confusion; I almost never sign into my account on the public registry, and didn't remember that there were two images.

**Comment by sudoforge** [comment]
>_< I can't believe I glossed over that in your initial comment.

**Comment by sudoforge** [comment]
@KetchupBomb Just an FYI, I've closed out #96, which talked about the decision regarding the names. The associated commit has an updated README which discusses the plan regarding deprecation of the `sudoforge/murmur` and `sudoforge/murmurd` image names.
